### PR TITLE
Enable MUSIC tests and correct NEST testsuite results summary

### PR DESCRIPTION
**Pull request by gtrensch**
This PR reactivates the MUSIC tests. It also fixes an issue with incorrect entries in the NEST test suite results table that were encountered when forcing an error condition to test the correct behavior of the test suite.

_Resolved issues:_ #1557 
_Suggested reviewer:_ @heplesser

### CODE REVIEW

File: testsuite/do_tests.sh [hunk 1]
```diff
@@ -346,101 +346,104 @@ else
...
+        # Calculate the total number of processes from the '.music' file.
+        np=$(($(sed -n 's/np=//p' ${music_file} | paste -sd'+' -)))
+        command="$(sli -c "${np} (${MUSIC}) (${test_name}) mpirun =")"
+
+        proc_txt="processes"
+        if test $np -eq 1; then proc_txt="process"; fi
+        echo          "Running test '${test_name}' with $np $proc_txt... " >> "${TEST_LOGFILE}"
+        printf '%s' "  Running test '${test_name}' with $np $proc_txt... "
+
+        # Copy everything to 'tmpdir'.
```
Comment (hakonsbm) [inline testsuite/do_tests.sh:381]:
```suggestion
        # Copy everything to 'tmpdir'.
```

File: testsuite/do_tests.sh [hunk 2]
```diff
@@ -346,101 +346,104 @@ else
...
+        TIME_TOTAL=$(( ${TIME_TOTAL} + ${TIME_ELAPSED} ))
+        sed -e 's/^/   > /g' ${TEST_OUTFILE} >> "${TEST_LOGFILE}"
+
+        # Retrieve the exit code. This is either the one of the mpirun call
+        # or of the accompanying shell script if present.
+        exit_code=$(cat exit_code)
+
+        # Count the total number of tests, the tests skipped, and the tests with error.
+        # The values will be stored in the XML report at 'junit_close'.
+        # Test failures and diagnostic information are also stored in the xml-report file
```
Comment (hakonsbm) [inline testsuite/do_tests.sh:410]:
```suggestion
        # Test failures and diagnostic information are also stored in the xml-report file
```

**Review by hakonsbm — CHANGES_REQUESTED** [review]
Thanks for fixing this! Looks good to me except for a couple lines with trailing whitespace.

**Review by hakonsbm — APPROVED** [review]

**Review by heplesser — APPROVED** [review]
Thanks, @gtrensch !

**Pull request merged**